source/android/adaptivecards/src/main/java/com/microsoft/adaptivecards/renderer/http/HttpRequestHelper.java

```diff
@@ -0,0 +1,113 @@
+        conn.setUseCaches(useCaches);
+
+        if (requestProperty != null)
+        {
+            // Set HTTP header
+            for (Map.Entry<String, String> entry : requestProperty.entrySet())
+            {
+                conn.setRequestProperty(entry.getKey(), entry.getValue());
+            }
+        }
+
+        return conn;
+    }
+
+    private static byte[] copyInputStreamToByteArray(InputStream inputStream)
+            throws IOException
+    {
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+        final int bufferSize = 4096;
+        byte[] buffer = new byte[bufferSize];
+        int size;
+        while ((size = inputStream.read(buffer)) != -1)
+        {
+            byteArrayOutputStream.write(buffer, 0, size);
+        }
+
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    public static byte[] get(String url, Map<String, String> requestProperty)
+            throws MalformedURLException, URISyntaxException, IOException
+    {
+        HttpURLConnection conn = connect(url, HTTP_METHOD_GET, requestProperty, false, true);
+        conn.connect();
+        BufferedInputStream inputStream = new BufferedInputStream(conn.getInputStream());
+        byte[] bytes = copyInputStreamToByteArray(inputStream);
+        inputStream.close();
+        return bytes;
+    }
+
+    public static byte[] get(String url)
+            throws MalformedURLException, URISyntaxException, IOException
+    {
+        return get(url, null);
+    }
+
+    private static  byte[] requestBody(String url, String method, Map<String, String> requestProperty, String body)
+            throws MalformedURLException, URISyntaxException, IOException
+    {
+        byte[] bodyBytes = body.getBytes("UTF-8");
+        requestProperty.put("Content-Length", String.valueOf(bodyBytes.length));
+        HttpURLConnection conn = connect(url, method, requestProperty, true, false);
+
+        OutputStream outputStream = conn.getOutputStream();
+        outputStream.write(bodyBytes);
+
+        BufferedInputStream inputStream = new BufferedInputStream(conn.getInputStream());
+        byte[] bytes = copyInputStreamToByteArray(inputStream);
+
+        outputStream.close();
+        inputStream.close();
+
+        return bytes;
+    }
+
+    public static byte[] post(String url, Map<String, String> requestProperty, String body)
+            throws MalformedURLException, URISyntaxException, IOException
+    {
+        return requestBody(url, HTTP_METHOD_POST, requestProperty, body);
+    }
+
+    public static byte[] put(String url, Map<String, String> requestProperty, String body)
+            throws MalformedURLException, URISyntaxException, IOException
+    {
+        return requestBody(url, HTTP_METHOD_PUT, requestProperty, body);
+    }
+
+    public static final String HTTP_METHOD_GET = "GET";
+    public static final String HTTP_METHOD_POST = "POST";
+    public static final String HTTP_METHOD_PUT = "PUT";
```

Comment:
It appears we will not support POST and PUT and it's the user that needs to build their own HTTP request.  This may be pulled in the future.

---
In reply to: [126766025](https://github.com/Microsoft/AdaptiveCards/pull/461#discussion_r126766025) [](ancestors = 126766025)